### PR TITLE
MERGE PRODUCTION ENVIRONMENT:

### LINKED ISSUE #49: Setup a production environment:
Setup a production environment including a reverse proxy with nginx.
We need to implement an Nginx service to efficiently serve both the backend and frontend of our project.
Additionally, we must define a clear production and development environments to simplify project deployment and testing.

### PR DESCRIPTION
**Pull request by riblanc**
**Description:**
This pull request introduces the necessary changes to add a production environment configuration to the project.
It includes updates to configuration files and Docker setups to ensure a seamless transition to production mode.

**Changes Made:**

1. **.env.sample**:
   - Added production-specific environment variables.
   - Updated API_URL and WS_BASE_URL to match production URLs.

2. **.gitignore**:
   - Added .env.dev to the gitignore file.

3. **back/Dockerfile**:
   - Added a new Docker build stage for production.
   - Copies the necessary production files and dependencies.
   - Sets NODE_ENV to production for the production image.

4. **back/entrypoint.sh**:
   - Updated to export NODE_ENV as production.

5. **front/Dockerfile**:
   - Added a new Docker build stage for production.
   - Copies the production build output.

6. **front/entrypoint.sh**:
   - Updated to export NODE_ENV as production.

7. **front/nuxt.config.ts**:
   - Updated the default API_URL and WS_BASE_URL for production.

8. **front/stores/auth.store.ts**:
   - Updated API_URL references to use the production URL.

9. **docker-compose.dev.yml**:
   - Added a development-specific Docker Compose file.

10. **docker-compose.yml**:
    - Updated Docker Compose file for production.
    - Added NGINX configuration for reverse proxying to the frontend and backend.

11. **proxy/templates/default.conf.template**:
    - Added NGINX configuration for reverse proxying to the frontend, backend, and WebSocket.

12. **proxy/entrypoint-wrapper.sh**:
    - Added a script to control NGINX start based on NODE_ENV.

**How to Test:**
1. Clone this branch and navigate to the project directory.
2. Run `docker-compose up --build` for production or `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up --build` for development.
3. Verify that the project runs without errors in both development and production environment.

Close #49

### CODE REVIEW

**Comment by AdrianWeaver** [comment]
I don't have the time to feedback this right now I'm sorry...
But I thought I heard you talk about changing the Makefile as well and I don't see any changes to it?

File: back/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,40 @@
-FROM node:20.5.1
+FROM node:20.5.1 as build_and_migrate
 
 WORKDIR /app
 
+ARG PORT=3000
+ARG JWT_SECRET="SUp3r_D3f4u1t_53cr3t"
+ARG WEB_APP_URL="http://localhost"
```
Comment (Elena-Lou) [inline back/Dockerfile:7]:
```suggestion

```

Comment (Elena-Lou) [reply back/Dockerfile:7]: We agreed during review that they were useless in the dockerfile

File: docker-compose.yml [hunk 2]
```diff
@@ -8,54 +8,81 @@ services:
...
+      target: prod
+      args:
+        - NODE_ENV=production
+    expose:
+      - 3000
     env_file:
       - .env
-    ports:
-      - "3000:3000"
+      - ./back/.env
```
Comment (Elena-Lou) [inline docker-compose.yml:39]:
```suggestion
```
The Database url being already in the environment from the dotenv, there is no need for the backend one

File: front/layouts/default.vue [hunk 3]
```diff
@@ -32,6 +32,7 @@ export default defineNuxtComponent({
     }
   },
   beforeMount() {
+    console.log('beforeMount: ', this.$config.app.API_URL);
```
Comment (Elena-Lou) [inline front/layouts/default.vue:35]:
```suggestion

```
Lonely debug

**Review by Elena-Lou — CHANGES_REQUESTED** [review]
Thank you for your work on this. The difference between "prod" and "dev" environment has always been very abstract to me but thanks to this, it is becoming clearer

**Review by Elena-Lou — APPROVED** [review]
Thank you for the updates and improvment

**Pull request merged**